fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -1229,30 +1235,14 @@ public FedoraResource getBaseVersion() {
     return null;
   }
 
-  @Override
-  public void enableVersioning() {
-        if (!isVersioned()) {
-            findOrCreateTimeMap();
-        }
-  }
-
-  @Override
-  public void disableVersioning() {
-        getTimeMap().delete();
-  }
-
     @Override
-    public boolean isVersioned() {
-        try {
-            return getNode(getNode(), LDPCV_TIME_MAP, false) != null;
-        } catch (final RepositoryException ex) {
-            throw new RepositoryRuntimeException(ex);
-        }
+    public boolean isOriginalResource() {
+        return !isAcl() && !isMemento() && !isTimeMap();
```

Comment:
It looks like FedoraTimeMap and FedoraWebacAcl have their own classes that inherit from FedoraResource. What would you think about overriding this method in those classes to just return `false`, and change this method to only check `!isMemento()`?